### DISCUSSION TITLE
Using VanUI without npm?

### DISCUSSION THREAD

**Discussion by b-rad-c** [vanjs-org/van#129] [Ideas]
I'm just getting started with VanJS, and so far I love the minimal reactive API, what drew me to VanJS is that it can be used without a build step or any additional dependencies. I think it would be great if the VanUI components could be used like that. I tried downloading [this file](https://github.com/vanjs-org/van/blob/main/components/dist/van-ui.js) and including it via html <script> tag but it produced the following error:

`SyntaxError: Unexpected identifier 'van'. import call expects one or two arguments.`

Is there any possibility of creating a release of VanUI that could be included via script tag? I'm pretty knowledgable on JS but not sure what would be involved in compiling this down to vanilla js.

**Comment by Tao-VanJS** [comment]
Currently VanUI only supports NPM. I can consider supporting other forms. But there are caveats:

* The library defines multiple UI components, but users might only want some of them. With NPM and bundling tools, the final bundle size can be trimmed down with tree-shaking. If I release VanUI as a standalone script to be imported in the browser, tree-shaking won't happen which leads to non-optimal total bundle size.
* VanUI depends on VanJS. If we publish VanUI as an ESM, the VanUI script needs to import VanJS. And the user likely wants to import VanJS for their app as well. The user needs to ensure they're importing exactly the same version of VanJS as VanUI does to avoid duplicate import and potentially conflicts between versions. If we publish VanUI as an inline script, the user needs to explicitly import VanJS script in the HTML file for VanUI to work properly, which is not quite ideal as the published VanUI script is not hermetic.

For those reasons, I feel that in the context of VanUI, NPM is a much better solution compared to other alternatives. That said, I will keep your request in mind while deciding whether to publish VanUI in other forms in the future.

**Comment by Tao-VanJS** [comment]
Since it's not hard to support using VanUI via a `<script>` tag, I just added the support. You can refer to https://github.com/vanjs-org/van/tree/main/components#via-a-script-tag for details.

Let me know how it works for your use cases :-)

**Comment by b-rad-c** [comment]
Awesome! Thanks for your quick reply and update. Will test it out in the
next few days and let you know!
